### PR TITLE
Changed Incorrect Heading Semantics

### LINKED ISSUE #8868: [a11y] Missing and incorrect heading semantics
On add-on pages, there are several headers (Screenshots, About this extension, Other users with this extension also installed, Rate your experience, Permissions, More information, Add to collection, Release notes for {version}) which are not exposed as headings for accessibility. Given the amount of information on these pages, this makes it very difficult for a screen reader user to jump to the information they're interested in. For example, a user often wants to quickly read "About this extension", but right now, a screen reader user can't jump there easily; they often end up laboriously cursoring line by line through the extension metadata, screenshots, etc. This makes for a pretty painful experience, to the point where I as a screen reader user tend to dread or even avoid searching for add-ons.

This can be made into a delightful experience as follows:

1. Make all elements with class `Card-header` into heading level 2. You can either use the h2 tag, or if you don't want to mess with tags for styling reasons, use `role="heading" aria-level="2"`.
2. Now that these main headings are h2, the content under "Other users with this extension also installed" needs to have its heading levels adjusted:
    - `.SearchResult-name` should be h3, not h2. Again, you can also use aria-level="3" if you prefer.
    - `.SearchResult-users` should be h4, not h3, if it needs to be a heading at all. Personally, I don't think this is worthy of a heading. I'd suggest changing this to a paragraph, either the `p` tag or `role="paragraph"`.

### PR DESCRIPTION
**Pull request by tharun208**
Fixes #8868 
  This fixes incorrect heading semantics
Fixes #8868 
 Changed the heading to the mentioned heading semantics in the issue description

### CODE REVIEW

File: src/ui/components/Card/index.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default class Card extends React.Component {
           'Card--no-footer': !footer,
         })}
       >
-        {header ? <header className="Card-header">{header}</header> : null}
+        {header ? <h2 className="Card-header">{header}</h2> : null}
```
[the remark this answers is not in the document]
Comment (muffinresearch) [reply src/ui/components/Card/index.js]: It's not too clear to me why `header` is become a heading here since presumably `header` is being used as a container?

[the remark this answers is not in the document]
Comment (tharun208) [reply src/ui/components/Card/index.js]: I fixed this but the build is failing, even though there are no linting issues

[the remark this answers is not in the document]
Comment (tharun208) [reply src/ui/components/Card/index.js]: I again converted into a header tag with the tags. But the build is failing

[the remark this answers is not in the document]
Comment (willdurand) [reply src/ui/components/Card/index.js]: @muffinresearch I think the idea of the OP is to mark these card "headers" as headings because in most cases, we inject textual content in the `header` prop (which is often the title of that card).

[the remark this answers is not in the document]
Comment (muffinresearch) [reply src/ui/components/Card/index.js]: If this goes in the direction of changing this to a heading element, we'll need to check that there isn't any props passing complete headings anywhere.

File: src/ui/components/Card/index.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ export default class Card extends React.Component {
           'Card--no-footer': !footer,
         })}
       >
-        {header ? <header className="Card-header">{header}</header> : null}
+        {header ? <header className="Card-header" role="heading" aria-level="2">{header}</header> : null}
```
Comment (willdurand) [inline src/ui/components/Card/index.js]: I believe we should revert this change for now.

**Comment by willdurand** [comment]
@tharun208 sorry for not getting back to that earlier. While I think the proposed changes are ok, I'd like to have a complete accessibility review so that we can make changes accordingly on all the pages at once, instead of trying to fix only a few things. I requested a review in https://github.com/mozilla/addons-frontend/issues/3111

**Comment by bobsilverberg** [comment]
@willdurand Do you think we should also discontinue work on https://github.com/mozilla/addons-frontend/pull/8939 for the same reason?

**Comment by willdurand** [comment]
> @willdurand Do you think we should also discontinue work on #8939 for the same reason?

probably yes
